marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid

```diff
@@ -0,0 +1,178 @@
+
+* [How Platform OS Works](/how-platformos-works)
+* [Get Started: Pages](/get-started/pages/pages)
+* [Using a Static AMP Template](/how-to/amp/using-static-amp-template)
+* [Converting AMP HTML Pages to Layouts and Pages](/how-to/amp/converting-amp-html-layouts-pages)
+ 
+## Steps
+Feeding data into AMP pages with GraphQL, in this case, displaying products on a page is a four-step process:
+
+1. Display list of products 
+2. Create JSON endpoint
+3. Fetch listing data
+4. Display product details
+
+### Step 1: Display list of products
+
+Create the main page, HTML extended with AMP tags and objects `marketplace_builder/views/pages/product-list.liquid`:
+
+<pre><code class="language-liquid">
+---
+slug: product-list
+format: html
+layout_name: application
+---
+<!-- AMP dependencies -->
+<script custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js" async=""></script>
+<script custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js" async=""></script>
+<script custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.1.js" async=""></script>
+..........
+  <amp-state id="products">
+    <script type="application/json">
+      {
+        "order": "desc"
+      }
+    </script>
+  </amp-state>
+
+  <select on="change: AMP.setState({products: {order: event.value}})">
+    <option value="desc">Price: High-Low</option>
+    <option value="asc">Price: Low-High</option>
+  </select>
+
+  <amp-list [src]="'api/products-' + products.order + '.json'" src="api/products-desc.json" layout="responsive">
+    <template type="amp-mustache">
+      <a href="/product-details/{% raw %}{{ slug }}{% endraw %}">
+        <div>
+          <amp-img src="{% raw %}{{image}}{% endraw %}" width="300" layout="responsive" noloading=""></amp-img>
+          <h2>{% raw %}{{name}}{% endraw %}</h2>
+          {% raw %}{{description}}{% endraw %}
+        </div>
+        <div>${% raw %}{{ price }}{% endraw %}</div>
+      </a>
+    </template>
+  </amp-list>
+</code></pre>
+
+#### Notes: 
+* This page declares the AMP state variable `products` with a JSON data structure format. 
+* `"order": "desc"` is the default sort order.
+* `<select>` changes the sort order stored in the state variable
+* The data source for `<amp-list>` is either `products-asc.json` or `products-desc.json` and is chosen based on the state variable `products.order`.
+* `amp-mustache` enables the 'mustache' (double curly braces) templating syntax.
+* The `{% raw %}` tag instructs the liquid preprocessor not to interpret the `{{}}` tags, and leaves them to AMP/mustache to process. 
+
+
+### Step 2: Create JSON endpoint 
+Create the JSON endpoint, a dynamically generated JSON document from which AMP fetches the data that it displays `mviews/pages/api/products-asc.json.liquid`: 
+
+<pre><code class="language-liquid">
+---
```

Comment:
raw / endraw needed for it to display in docs

Reply:
Thanks! Fixed.